pkg/kubelet/kubelet_nodecache.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"context"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+// GetCachedNode compares the currently cached node to the node in the informer,
+// and returns the newer one.
+func (kl *Kubelet) GetCachedNode() (*v1.Node, error) {
+	informerNode, err := kl.GetNode()
+	if informerNode == nil || err != nil {
+		if kl.cachedNode != nil {
+			klog.Errorf("failed to list node; using cached node: %v", err)
+			return kl.cachedNode, nil
+		}
+		return kl.initialNode(context.TODO())
+	}
+
+	if kl.cachedNode == nil {
+		kl.cachedNode = informerNode
+		return informerNode, nil
+	}
+
+	isNewer, err := isNewer(informerNode, kl.cachedNode)
+	if err != nil {
+		// In error cases, default to the informer node.
+		klog.Errorf("failed to check if node is newer; using informer node: %v", err)
+		kl.cachedNode = informerNode
+	}
+	if kl.cachedNode == nil || isNewer {
+		kl.cachedNode = informerNode
+	}
+	return kl.cachedNode, nil
+}
+
+// GetNodeSync forces a refresh of the cache by making a synchronous call to the API server
+// for the most recent node info.
+func (kl *Kubelet) GetNodeSync() (*v1.Node, error) {
+	if kl.kubeClient == nil {
+		return kl.initialNode(context.Background())
+	}
+	node, err := kl.kubeClient.CoreV1().Nodes().Get(context.Background(), string(kl.nodeName), metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	kl.cachedNode = node
+	return node, nil
+}
+
+// isNewer checks if the informer node is newer than the cached node based on the ResourceVersion.
+func isNewer(informerNode, cachedNode *v1.Node) (bool, error) {
```

Comment:
nit: short circuit if `informerNode.ResourceVersion == cachedNode.ResourceVersion`

Reply:
Actually, you might be able to just check for pointer equality here `informerNode == cachedNode`

Reply:
done!